### ISSUE TITLE
Phases are not disjoint. The following parameters are specified in multiple phases:

### ISSUE THREAD

**Issue by quancs** [speediedan/finetuning-scheduler#3]
## 🐛 Bug

Hello, thanks for making this nice callback. 
I'm trying to use `finetuning-scheduler` to train my model phase-by-phase: phase 1, I want to first train part of my model; phase 2, I want to train the whole model. For the following config, the callback reports the error: `Phases are not disjoint`. So, I want to know how could I acheive my goal by using this callback.

```yaml
0:
  params: # the parameters for each phase definition can be fully specified
  - model.enc
  - model.dec
  - model.masker1
  - model.mvdr
  max_transition_epoch: 30
  lr: 0.001 # per-phase maximum learning rates can be specified
1:
  params:
  - model
  lr: 0.001 # per-phase maximum learning rates can be specified
```

### Environment

- Fine-Tuning Scheduler Version (e.g., 0.1.0):
- PyTorch Lightning Version (e.g., 1.5.0):
- PyTorch Version (e.g., 1.10):
- Python version (e.g., 3.9):
- OS (e.g., Linux):
- CUDA/cuDNN version:
- GPU models and configuration:
- How you installed PyTorch (`conda`, `pip`, source):
- If compiling from source, the output of `torch.__config__.show()`:
- Any other relevant information:

**Comment by quancs** [comment]
And I have another question: will the trainer use a new optimizer in a new phase?

**Comment by quancs** [comment]
I see it's gradual unfreezing the models...
